core/src/main/java/com/linecorp/armeria/server/streaming/JsonLines.java

```diff
@@ -0,0 +1,395 @@
+        return HttpResponse.of(sanitizeHeaders(headers), toHttpData(contentConverter, content), trailers);
+    }
+
+    private static ResponseHeaders sanitizeHeaders(ResponseHeaders headers) {
+        if (headers == defaultHttpHeaders) {
+            return headers;
+        }
+        return ensureContentType(ensureHttpStatus(headers));
+    }
+
+    static ResponseHeaders ensureHttpStatus(ResponseHeaders headers) {
+        final HttpStatus status = headers.status();
+        if (status.equals(HttpStatus.OK)) {
+            return headers;
+        }
+
+        if (!warnedStatusCode) {
+            logger.warn(
+                    "Overwriting the HTTP status code from '{}' to '{}' for JSON Lines. " +
+                            "Do not set an HTTP status code on the HttpHeaders when calling factory methods in '{}', " +
+                            "or set '{}' if you want to specify its status code. " +
+                            "Please refer to https://jsonlines.org/ for more information.",
+                    status, HttpStatus.OK, JsonLines.class.getSimpleName(), HttpStatus.OK);
+            warnedStatusCode = true;
+        }
+        return headers.toBuilder().status(HttpStatus.OK).build();
+    }
+
+    static ResponseHeaders ensureContentType(ResponseHeaders headers) {
+        final MediaType contentType = headers.contentType();
+        if (contentType == null) {
+            return headers.toBuilder()
+                    .add(HttpHeaderNames.CONTENT_TYPE, MediaType.JSON_LINES.toString())
+                    .build();
+        }
+
+        if (contentType.is(MediaType.JSON_LINES)) {
+            return headers;
+        }
+
+        if (!warnedContentType) {
+            logger.warn("Overwriting content-type from '{}' to '{}' for JSON Lines. " +
+                            "Do not set a content-type on the HttpHeaders when calling factory methods in '{}', " +
+                            "or set '{}' if you want to specify its content-type. " +
+                            "Please refer to https://jsonlines.org/ for more information.",
+                    contentType, MediaType.JSON_LINES,
+                    JsonLines.class.getSimpleName(), MediaType.JSON_LINES);
+            warnedContentType = true;
+        }
+        return headers.toBuilder()
+                .contentType(MediaType.JSON_LINES)
+                .build();
+    }
+
+    private static HttpData toHttpData(ObjectMapper mapper, @Nullable Object value) {
+        try {
+            final ByteArrayOutputStream out = new ByteArrayOutputStream();
+            // If the mapper is same don't need disable indentation.
+            if (mapper == defaultMapper) {
+                mapper.writeValue(out, value);
+            } else {
+                mapper.disable(SerializationFeature.INDENT_OUTPUT);
+                JsonNode root = mapper.valueToTree(value);
+                mapper.writeValue(out, root);
```

Comment:
We should not mutate user-defined `mapper`.
```suggestion
                JsonNode root = mapper.valueToTree(value);
                defaultMapper.writeValue(out, root);
```